python/GafferOSL/FocalBlur.gfr

```diff
@@ -0,0 +1,889 @@
+__children["FocalBlur"]["Expression7"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
+__children["openColorIO"] = GafferImage.OpenColorIOConfigPlug( "openColorIO", flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, )
+parent.addChild( __children["openColorIO"] )
+__children["defaultFormat"] = GafferImage.FormatPlug( "defaultFormat", defaultValue = GafferImage.Format( 1920, 1080, 1.000 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, )
+parent.addChild( __children["defaultFormat"] )
+parent["frame"].setValue( 100.0 )
+parent["variables"]["imageCataloguePort"]["value"].setValue( 38889 )
+Gaffer.Metadata.registerValue( parent["variables"]["imageCataloguePort"], 'readOnly', True )
+Gaffer.Metadata.registerValue( parent["variables"]["projectName"]["name"], 'readOnly', True )
+Gaffer.Metadata.registerValue( parent["variables"]["projectRootDirectory"]["name"], 'readOnly', True )
+Gaffer.Metadata.registerValue( parent["variables"]["renderPass"]["name"], 'readOnly', True )
+Gaffer.Metadata.registerValue( parent["variables"]["renderPass"]["value"], 'plugValueWidget:type', 'GafferSceneUI.RenderPassEditor._RenderPassPlugValueWidget' )
+Gaffer.Metadata.registerValue( __children["FocalBlur"], 'uiEditor:emptySectionIndices', IECore.IntVectorData( [  ] ) )
+Gaffer.Metadata.registerValue( __children["FocalBlur"], 'layout:section:Settings.Quality:collapsed', False )
+Gaffer.Metadata.registerValue( __children["FocalBlur"], 'layout:activator:cameraNotScene', 'parent["cameraMode"].getValue() != "scene"' )
+Gaffer.Metadata.registerValue( __children["FocalBlur"], 'layout:activator:cameraScene', 'parent["cameraMode"].getValue() == "scene"' )
+Gaffer.Metadata.registerValue( __children["FocalBlur"], 'layout:activator:cameraCustom', 'parent["cameraMode"].getValue() == "custom"' )
+Gaffer.Metadata.registerValue( __children["FocalBlur"], 'uiEditor:emptySections', IECore.StringVectorData( [  ] ) )
+Gaffer.Metadata.registerValue( __children["FocalBlur"], 'layout:activator:depthInterpretationFilteredWithBackground', 'parent["depthInterpretation"].getValue() == "filteredWithBackground"' )
+Gaffer.Metadata.registerValue( __children["FocalBlur"]["user"], 'layout:section', 'User' )
+Gaffer.Metadata.registerValue( __children["FocalBlur"]["in"], 'nodule:type', 'GafferUI::StandardNodule' )
+Gaffer.Metadata.registerValue( __children["FocalBlur"]["in"], 'description', 'The input image' )
+Gaffer.Metadata.registerValue( __children["FocalBlur"]["in"], 'plugValueWidget:type', '' )
+Gaffer.Metadata.registerValue( __children["FocalBlur"]["in"], 'noduleLayout:spacing', 2.0 )
+Gaffer.Metadata.registerValue( __children["FocalBlur"]["in"], 'layout:section', 'Settings' )
+Gaffer.Metadata.registerValue( __children["FocalBlur"]["in"], 'layout:index', 0 )
+__children["FocalBlur"]["out"].setInput( __children["FocalBlur"]["BoxOut"]["__out"] )
```

Comment:
We're somehow serialising these `setInput()` calls, I noticed it when loading a script saved with the previous iteration of FocalBlur and saw errors due to the removal of the BoxOuts in this update.

Reply:
This seems to be a general problem with ExtensionAlgo based nodes. I see the same problem with MetadataOverlay.

Reply:
Thanks, after discussing offline we're happy to punt on this for now. This will need to be more generally solved at a later date.